android/java/org/chromium/chrome/browser/crypto_wallet/activities/AccountSelectorActivity.java

```diff
@@ -92,4 +70,39 @@ public void onAccountClick(WalletListItemModel walletListItemModel) {
             }
         }
     }
+
+    @SuppressLint("NotifyDataSetChanged")
+    private void initAccounts() {
+        mKeyringModel.getAccounts(accountInfos -> {
+            mAccountInfos = accountInfos;
+            List<WalletListItemModel> walletListItemModelList = new ArrayList<>();
+            for (AccountInfo accountInfo : mAccountInfos) {
+                if (!accountInfo.isImported) {
+                    walletListItemModelList.add(
+                            new WalletListItemModel(R.drawable.ic_eth, accountInfo.name,
+                                    accountInfo.address, null, null, accountInfo.isImported));
+                }
+            }
+
+            mWalletCoinAdapter.setWalletListItemModelList(walletListItemModelList);
+            mWalletCoinAdapter.notifyDataSetChanged();
+
+            mKeyringModel.mSelectedAccount.observe(this, selectedAccountInfo -> {
+                if (selectedAccountInfo == null) return;
+
+                mWalletCoinAdapter.updateSelectedNetwork(
+                        selectedAccountInfo.name, selectedAccountInfo.address);
+                boolean callDataSetChanged = true;
+                for (AccountInfo accountInfo : mAccountInfos) {
+                    if (selectedAccountInfo.address.equals(accountInfo.address)) {
+                        callDataSetChanged = false;
+                        break;
+                    }
+                }
+                if (callDataSetChanged) {
+                    initAccounts();
+                }
```

Comment:
nit: just a note, this workaround can be avoided with [mAccountAllAcount](https://github.com/brave/brave-core/blob/4320c8a22c9d44947bc9eea5a3c5059b364fe004/android/java/org/chromium/chrome/browser/crypto_wallet/activities/BuySendSwapActivity.java#L1564) so either can copy or will improve it later